broker/src/main/java/org/apache/rocketmq/broker/processor/PopMessageProcessor.java

```diff
@@ -350,7 +353,7 @@ private RemotingCommand processRequest(final Channel channel, RemotingCommand re
         if (requestHeader.isOrder()) {
             reviveQid = KeyBuilder.POP_ORDER_REVIVE_QUEUE;
         } else {
-            reviveQid = randomQ % this.brokerController.getBrokerConfig().getReviveQueueNum();
+            reviveQid = Math.abs(ckMessageNumber.getAndIncrement() % this.brokerController.getBrokerConfig().getReviveQueueNum());
```

Comment:
Whether the range of AtomicInteger is a bit small, even if there is abs, it may return a negative number, and the message volume may be out of bounds if it is too large. Is AtomicLong or ThreadLocalRandom better?

Reply:
I think atomicLong can be better and I will update it.